### DISCUSSION TITLE
Error: Insufficient information provided for naming the alarms and/or monitoring section: Please provide alarmFriendlyName, humanReadableName, or namedConstruct as a fallback

### DISCUSSION THREAD

**Discussion by bestickley** [cdklabs/cdk-monitoring-constructs#402] [Q&A]
I got the error: `Error: Insufficient information provided for naming the alarms and/or monitoring section: Please provide alarmFriendlyName, humanReadableName, or namedConstruct as a fallback` when deploying this [Nextjs](https://constructs.dev/packages/cdk-nextjs-standalone/v/3.2.1) construct and I'm not sure why. Can anyone help me understand how I can resolve the issue?

**Comment by mobob** [comment]
I just came across this lib, and first try i hit this too. It probably depends on what service you're using, but fwiw i'm using app sync, and just had to add in an `alarmFriendlyName` to the constructor to resolve the synth issue:

```
    monitoring
      .addLargeHeader('Backend Monitoring')
      .monitorAppSyncApi({ api: appSyncApp, alarmFriendlyName: 'fsta-backend' });
```

**Comment by ghost** [comment]
Hey there! Apologies on the very late reply

The error Insufficient information error is thrown here: https://github.com/cdklabs/cdk-monitoring-constructs/blob/2357fe09f9d9af7ea1c4bc5b53b3f7b9c0e8ae24/lib/dashboard/MonitoringNamingStrategy.ts#L87-L90

...via: https://github.com/cdklabs/cdk-monitoring-constructs/blob/2357fe09f9d9af7ea1c4bc5b53b3f7b9c0e8ae24/lib/dashboard/MonitoringNamingStrategy.ts#L54-L56

From what I remember, there [are requirements](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-requirements) on what values we can use since they can end up in CloudFormation parameters- providing the `alarmFriendlyName` like @mobob mentions helps us generate a string withing the Cfn requirements. Hope this helps!

*CC @echeung-amzn to correct me if I'm wrong :P*

**Reply by ghost** [reply]
That's right. Like @mobob mentioned as well, it can depend on what service you're using and whether the props you're passing in end up being sufficient enough to form a decently recognizable/unique name or not.

@bestickley I'm not familiar with the NextJS construct, does the error happen to spit out more context on what it's failing around? Are you using `monitorScope`?

**Reply by bestickley** [reply]
@echeung-amzn and @cayman-amzn, thank you. I'm guessing it's an issue with the name as you suggested. I'll test this out when I can find time. See [here](https://github.com/awslabs/green-boost/blob/main/packages/gboost/templates/crud-core/infra/src/app/stateless/monitor-stack.ts#L53-L72) as example for how I'm using construct.
